### PR TITLE
Deprecate `Deployment` class and deployment `build` and `apply` commands

### PR DESCRIPTION
**Pull request by desertaxle**
Adds a deprecation warning to the `Deployment` class along with the the deployment `build` and `apply` commands in lieu of `flow.deploy` and `prefect deploy`. Adds some logic to enable deprecation of specific commands in addition to full command groups.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: tests/test_deployments.py [hunk 1]
```diff
@@ -72,6 +73,19 @@ async def ensure_default_agent_pool_exists(session):
...
+    with pytest.warns(
+        PrefectDeprecationWarning,
+        match=(
+            "prefect.deployments.deployments.Deployment has been deprecated."
+            " It will not be available after Sep 2024."
+            " Use `flow.deploy` to deploy your flows instead."
+            " Refer to the upgrade guide for more information"
+        ),
+    ):
+        Deployment(name="foo")
```
Comment (serinamarie) [inline tests/test_deployments.py:86]: 🤯

File: src/prefect/deployments/deployments.py [hunk 2]
```diff
@@ -320,13 +323,20 @@ def load_deployments_from_yaml(
...
+    start_date="Mar 2024",
+    help="Use `flow.deploy` to deploy your flows instead."
+    " Refer to the upgrade guide for more information:"
+    " https://docs.prefect.io/latest/guides/upgrade-guide-agents-to-workers/.",
 )
 class Deployment(BaseModel):
     """
+    DEPRECATION WARNING:
+
+    This class is deprecated as of version March 2024 and will not be available after September 2024.
```
Comment (serinamarie) [inline src/prefect/deployments/deployments.py:336]:
```suggestion
    This class is deprecated as of March 2024 and will not be available after September 2024.
```

File: src/prefect/deployments/deployments.py [hunk 3]
```diff
@@ -320,13 +323,20 @@ def load_deployments_from_yaml(
...
+    It has been replaced by the `flow.deploy`, which offers enhanced functionality and better a better user experience.
```
Comment (serinamarie) [inline src/prefect/deployments/deployments.py:337]:
```suggestion
    It has been replaced by `flow.deploy`, which offers enhanced functionality and better a better user experience.
```

**Review by serinamarie — APPROVED** [review]
Wow I didn't really realize these would be deprecated as well. Would it make sense to add a blurb in the description on the motivation? I feel like this will be a frequently viewed PR 👀 and it could be useful for anyone wondering about it.

**Comment by desertaxle** [comment]
> Wow I didn't really realize these would be deprecated as well. Would it make sense to add a blurb in the description on the motivation? I feel like this will be a frequently viewed PR 👀 and it could be useful for anyone wondering about it.

We'll have an explanation in the release notes tomorrow that will cover the motivation behind these deprecations. I updated the description of this PR slightly, but I can add more detail tomorrow while working on a release notes highlight.

**Comment by serinamarie** [comment]
Awesome!

**Pull request merged**